Lexis+.js

```diff
@@ -0,0 +1,215 @@
+
+  Zotero is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+  ***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, _url) {
+	if (doc.title.match(/.*results.*/)) {
+		return "multiple";
+	}
+	else if (doc.title.match(/[a-zA-Z. ]+\s§\s\d+/)
+           || doc.title.match(/act/i)
+           || doc.title.match(/p\.l\./i)) { // Match: ... Tex. Bus. & Com. Code § 26.01 ...
+		return "statute";
+	}
+	else if (doc.title.match(/\d+\s[a-zA-Z0-9. ]+\s\d+/)) { // Match: ... 5 U.S. 137 ...
+		return "case";
+	}
+	// TODO secondary sources
+
+	return false;
+}
+
+function getSearchResults(doc, url) {
+	var items = {};
+	var nextTitle;
+
+	if (detectWeb(doc, url) == "multiple") {
+		// TODO check what type of element it is (currently only working for 'cases' searches)
+		var titles = doc.querySelectorAll('a.titleLink');
+		var dates = doc.querySelectorAll('span.metaDataItem'); // Not technically only dates, but that's all I use it for atm
+		var nextDate;
+		var dateOffset = 1;
+    
+		// dates[0] is first court name
+		nextDate = dates[dateOffset];
+		dateOffset += 3;
+		// dates[2] is first citation
+		for (var i = 0; i < titles.length; i++) {
+			nextTitle = titles[i];
+			items[nextTitle.href] = nextTitle.textContent + "(" + nextDate.textContent + ")";
+      
+			// dates[0] is court name
+			nextDate = dates[dateOffset];
+      
+			// dates[2] is a citation
+		}
+  
+		return items;
+	}
+
+	return false;
+}
+
+async function doWeb(doc, url) {
+	if (detectWeb(doc, url) == 'multiple') {
+		let items = await Zotero.selectItems(getSearchResults(doc, url));
+		if (!items) return;
+		for (let url of Object.keys(items)) {
+			await scrape(await requestDocument(url));
+		}
+	}
+	else {
+		await scrape(doc, url);
+	}
+}
+
+async function scrape(doc, url) {
+	if (detectWeb(doc, url) == "case") {
+		var newCase = new Zotero.Item("case");
+		newCase.url = doc.location.href;
+    
+		newCase.title = text(doc, 'h1#SS_DocumentTitle');
+
+		var citation = text(doc, 'span.active-reporter');
+		newCase.reporterVolume = citation.substring(0, citation.indexOf(' '));
+		newCase.reporter = citation.substring(citation.indexOf(' ') + 1, citation.lastIndexOf(' '));
+		newCase.firstPage = citation.substring(citation.lastIndexOf(' ') + 1);
+
+		newCase.court = text(doc, 'p.SS_DocumentInfo', 0);
+
+		newCase.dateDecided = text(doc, 'span.date');
+
+		var docket = text(doc, 'p.SS_DocumentInfo', 2);
+		if (docket.match(/^no\./i)
+      || docket.match(/^\d+/)
+    || docket.match(/^case no\./i)) newCase.docketNumber = docket; // This won't be in perfect cite form, shouldn't be a hassle unless you're citing dozens of memorandum opinions
```

Comment:
spaces → tabs, and please use braces for all multi-line conditionals. (The weirdness of this structure is avoiding a lint warning, but it's still a multi-line conditional.)

Reply:
Done!